lib/porky_lib/file_service.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require 'aws-sdk-s3'
+require 'singleton'
+
+class PorkyLib::FileService
+  include Singleton
+
+  def read(bucket_name, file_key, options = {})
+    tempfile = Tempfile.new
+    s3.bucket(bucket_name).object(file_key).download_file(tempfile.path, options)
```

Comment:
Would it be worth it to separate out the File read/write functionality from the s3 download/upload related functionality? It could be nice to have a generic solution for both these scenarios

Reply:
The requirement was to have porky_lib to read/write to S3. We do not have any requirements to read/write to/from local storage as Heroku's filesystem is ephemeral. We are storing reports/report data in S3.

Reply:
Makes sense 👍